src/transaction/log_append.cpp

```diff
@@ -185,7 +185,10 @@ LOG_RESET_APPEND_LSA (const LOG_LSA *lsa)
   log_Gl.hdr.append_lsa = *lsa;
   log_Gl.prior_info.prior_lsa = *lsa;
 #if defined (SERVER_MODE)
-  log_Gl.get_log_prior_sender ().reset_unsent_lsa (*lsa);
+  if (is_active_transaction_server () || is_page_server ())
```

Comment:
```suggestion
  if (!is_passive_transaction_server ())
```
Nit. I think it can be shortened

Reply:
I have considered this and `if log prior sender is initialized or not` as well, but I have decided to use this. What you suggested has the assumption that `NOT PTS -> ATS or PS`, which is actually true probably always, but I think it's better to avoid any assumption if it doesn't cost a lot.

Reply:
OK, I agree. Clearly stating a condition without inferring would likely make it easier to understand.

Reply:
This change here mandates a corresponding assert in `prior_sender::reset_unsent_lsa`:
```
  assert (is_active_transaction_server () || is_page_server ());
```